grails-app/controllers/org/broadinstitute/orsp/api/NewConsentGroupController.groovy

```diff
@@ -199,4 +200,22 @@ class NewConsentGroupController extends AuthenticatedController {
         }
         render(response)
     }
+
+
+    /**
+     * This action approve the link between a project and the consent.
+     *
+     */
+    def approveLink () {
+        try {
+            queryService.updateCollectionLinkStatus(params.consentKey, params.projectKey, IssueStatus.Approved.name)
+            response.status = 200
+        } catch (Exception e) {
+            response.status = 500
+            log.error("Exception updating collection links status: " + e)
+            flash.error = "Error updating collection links status: " + e
+        }
+        redirect(controller: "project", action: "main", params: [projectKey: params.projectKey, tab: "consent-groups"])
```

Comment:
I don't think we should be redirecting here (or anywhere, for that matter). If we redirect, that means we lose all context and state from the page that this was called from. The way our APIs should work is that they perform some kind of function, then return the result of that function to the React component that made the call. The component then reads that information and performs state updates as needed. We're trying to get away from the multi-page pattern and moving toward the single-page pattern.

Reply:
I think I see why you're doing this ... and it has to do with the fact that the caller is in a GSP (see  `grails-app/views/consentGroup/list.gsp`). I'm not sure what the best tactical approach is here - maybe that page needs to go through the react conversion first, or we can add this bit of tech debt to handle in whatever story does handle this.

Reply:
I agree, when I moved the component to react I forgot to remove the redirection. I'll do that